### PR TITLE
[release-1.7] Cache the http client in order to reuse the connection 

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
manually cherry pick #27864

fixes: #27983

### CODE REVIEW

File: pilot/cmd/pilot-agent/status/server.go [hunk 1]
```diff
@@ -414,16 +426,8 @@ func (s *Server) handleAppProbe(w http.ResponseWriter, req *http.Request) {
...
-	// Construct a request sent to the application.
-	httpClient := &http.Client{
-		Timeout: time.Duration(prober.TimeoutSeconds) * time.Second,
-		// We skip the verification since kubelet skips the verification for HTTPS prober as well
-		// https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-		},
-	}
+	// get the http client must exist because
```
Comment (JimmyCYJ) [inline pilot/cmd/pilot-agent/status/server.go:429]: Update the comment? What if the connection initiated by a cached client is broken, does the client automatically reconnect?

Comment (JimmyCYJ) [reply pilot/cmd/pilot-agent/status/server.go:429]: @hzxuzhonghu could you please update the PR?

Comment (hzxuzhonghu) [reply pilot/cmd/pilot-agent/status/server.go:429]: @JimmyCYJ Yes, httpClient will automatically establish a connection if no connection available when send a request

Comment (hzxuzhonghu) [reply pilot/cmd/pilot-agent/status/server.go]: @JimmyCYJ Done

**Review by JimmyCYJ — APPROVED** [review]

**Pull request merged**